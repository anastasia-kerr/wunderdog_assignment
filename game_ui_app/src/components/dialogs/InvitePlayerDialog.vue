<template>
  <v-dialog :width="dialogSize" v-model="dialog">
    <template v-slot:activator="{ on, attrs }">
        <v-btn
          color="primary"
          dark
          v-bind="attrs"
          v-on="on"
          text
          v-text="headerLabel"
        >
        </v-btn>
      </template>
      <v-card>
        <v-card-title v-text="headerLabel" />
        <v-card-subtitle v-text="subheaderLabel"/>
        <v-card-text>
          <v-text-field :label="urlLabel" outlined readonly v-model="url" />
        </v-card-text>
        <v-card-actions>
          <v-spacer />
          <v-btn color="primary" v-text="btnLabel" @click="onBtnPress" />
        </v-card-actions>
      </v-card>
  </v-dialog>
</template>

<script lang="ts">
import Vue from 'vue'
import config from '../../configs'
const { smallDialogSize } = config

export default Vue.extend({
  name: 'InvitePlayers',

  data: () => ({
    dialog: false,
    dialogSize: smallDialogSize,
    urlLabel: 'Game\'s url',
    headerLabel: 'Invite player',
    subheaderLabel: 'Send the link below to your friend to join the game',
    btnLabel: 'Copy invitation link'
  }),
  computed: {
    url ():string {
      return `${location.origin}/game/${this.$route.params.id}`
    }
  },
  methods: {
    onBtnPress (): void {
      navigator.clipboard.writeText(this.url)
      this.dialog = false
    }
  }
})
</script>
