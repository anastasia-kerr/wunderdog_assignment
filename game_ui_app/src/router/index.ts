import Vue from 'vue'
import VueRouter, { RouteConfig } from 'vue-router'
import HomeView from '../views/HomeView.vue'
import GamePlayView from '../views/GamePlayView.vue'
import CreateGame from '../components/CreateGame.vue'

Vue.use(VueRouter)

const routes: Array<RouteConfig> = [
  {
    path: '/',
    name: 'home',
    component: HomeView
  },
  {
    path: '/game/:id',
    name: 'game',
    component: GamePlayView
  },
  {
    path: '/create-game',
    name: 'create-game',
    component: CreateGame
  }
]

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes
})

export default router
