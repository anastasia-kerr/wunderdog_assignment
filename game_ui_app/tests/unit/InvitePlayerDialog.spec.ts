import Vue from 'vue'
import InvitePlayerDialog from '@/components/dialogs/InvitePlayerDialog.vue'
import { shallowMount, createLocalVue, Wrapper } from '@vue/test-utils'
import Vuetify from 'vuetify'
import router from '../../src/router'

const localVue = createLocalVue()
Vue.use(Vuetify)

describe('InvitePlayerDialog', () => {
  let vuetify: Vuetify
  let wrapper:Wrapper<Vue>
  beforeEach(() => {
    vuetify = new Vuetify()
    wrapper = shallowMount(InvitePlayerDialog, {
      router,
      localVue,
      vuetify
    })
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
