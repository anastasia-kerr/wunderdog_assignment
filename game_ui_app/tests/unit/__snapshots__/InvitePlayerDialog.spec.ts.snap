// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitePlayerDialog renders correctly 1`] = `
<v-dialog-stub
  closedelay="0"
  contentclass=""
  opendelay="0"
  openonclick="true"
  origin="center center"
  retainfocus="true"
  transition="dialog-transition"
  width="320"
>
   
  <v-card-stub
    loaderheight="4"
    tag="div"
  >
    <v-card-title-stub
      tag="div"
    >
      Invite player
    </v-card-title-stub>
     
    <v-card-subtitle-stub
      tag="div"
    >
      Send the link below to your friend to join the game
    </v-card-subtitle-stub>
     
    <v-card-text-stub
      tag="div"
    >
      <v-text-field-stub
        backgroundcolor=""
        clearicon="$clear"
        errorcount="1"
        errormessages=""
        label="Game's url"
        loaderheight="2"
        messages=""
        outlined="true"
        readonly="true"
        rules=""
        successmessages=""
        type="text"
        value="http://localhost/game/undefined"
      />
    </v-card-text-stub>
     
    <v-card-actions-stub
      tag="div"
    >
      <v-spacer-stub
        tag="div"
      />
       
      <v-btn-stub
        activeclass=""
        color="primary"
        tag="button"
        type="button"
      >
        Copy invitation link
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-dialog-stub>
`;
